### PR TITLE
fix: sync conflict - check server value for null before decyption

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- fix for exception caused when server value is null
**- How I did it**
- added a check in sync conflict to check the server value for null/empty before decrypting.
**- How to verify it**
- run the functional test [atclient_sync_conflict_test.dart](https://github.com/atsign-foundation/at_client_sdk/compare/trunk...sync_conflict_decryption_value_check#diff-da7a6aa9c49b0c15a4bae124b9490c92e993e2f1ec6156292f093bed40a4bc40)

Fixes the exception reported in bug https://github.com/atsign-foundation/at_client_sdk/issues/742

### CODE REVIEW

File: at_functional_test/test/atclient_sync_conflict_test.dart [hunk 1]
```diff
@@ -33,6 +32,31 @@ void main() async {
     }
     await Future.delayed(Duration(seconds: 10));
   });
+
+  // server value is null, client has non value for same key.
```
Comment (gkc) [inline at_functional_test/test/atclient_sync_conflict_test.dart]: should be `client has non-null value`

File: at_functional_test/test/atclient_sync_conflict_test.dart [hunk 2]
```diff
@@ -33,6 +32,31 @@ void main() async {
...
+  // you should get a sync conflict populated with no exception thrown.
+  test('server value is null', () async {
```
Comment (gkc) [inline at_functional_test/test/atclient_sync_conflict_test.dart]: The test is good. However, it did take me quite a few minutes of reading and re-reading the code before I could figure out what was going on. Please can you add some comments to explain what is going on?

**Review by gkc — CHANGES_REQUESTED** [review]

File: at_functional_test/test/atclient_sync_conflict_test.dart (hunk 2 above)
Comment (purnimavenkatasubbu) [reply at_functional_test/test/atclient_sync_conflict_test.dart]: Updated the comments and pushed the changes.

File: at_functional_test/test/atclient_sync_conflict_test.dart (hunk 1 above)
Comment (purnimavenkatasubbu) [reply at_functional_test/test/atclient_sync_conflict_test.dart]: Done

**Review by gkc — APPROVED** [review]

**Comment by gkc** [comment]
Nice work, @murali-shris and @purnimavenkatasubbu

**Pull request merged**